lib/metrician/reporters/memcache.rb

```diff
@@ -16,7 +16,7 @@ def client_classes
       [
         self.class.memcached_gem? && Memcached,
         self.class.dalli_gem? && Dalli::Client,
-      ].compact
+      ].select{|x| x}
```

Comment:
I don't know that I like any of these, really. I'm not sure it's worth it, but I think what I'd prefer is to do a slight abstraction here so dealing with two classes wasn't quite so unwieldy.

Thinking about it more I could probably also be on board making two subclasses of this class that abstract out the differences.

Overall though I think I lean towards leaving it as is and shipping. We've already spent more time on this than I'd hoped.

I'm definitely on the fence though, so I could be persuaded either way. How do you feel about the above options?